src/smt/timeout_core_manager.h

```diff
@@ -69,52 +70,93 @@ class TimeoutCoreManager : protected EnvObj
  public:
   TimeoutCoreManager(Env& env);
 
-  /** get timeout core for the current set of assertions stored in ppAsserts.
+  /**
+   * Get timeout core for the current set of assertions stored in ppAsserts.
+   *
+   * Returns a pair containing a result and a list of formulas C. For details,
+   * see Solver::getTimeoutCore.
+   *
+   * If hasAssumptions is true, the timeout core is a subset of formulas in
+   * assumptions.
+   *
+   * Otherwise, the timeout core is a subset of formulas in ppAsserts.
    *
-   * Returns a pair containing a result and a list of formulas C. If the result
-   * is unknown and the reason is timeout, then the list C corresponds to
-   * a subset of the current assertions that cause a timeout in the specified
-   * by timeout-core-timeout. If the result is unsat, then C is an unsat core
-   * for the set of assertions. Otherwise, the list of formulas is empty and the
-   * result has the same guarantees as a response to checkSat.
+   * @param ppAsserts The preprocessed assertions
+   * @param ppSkolemMap Mapping from indices in ppAsserts to skolem it defined,
+   * if applicable.
+   * @param assumptions The assumptions
+   * @param hasAssumptions Whether the user provided assumptions.
```

Comment:
How is this different from `!assumptions.empty()`?

Reply:
See above, will try to simplify.